parcels/particleset.py

```diff
@@ -878,8 +878,10 @@ def execute(self, pyfunc=AdvectionRK4, pyfunc_inter=None, endtime=None, runtime=
             runtime = runtime.total_seconds()
         if isinstance(dt, delta):
             dt = dt.total_seconds()
-        if dt > 0 and dt <= 1e-6:
+        if abs(dt) <= 1e-6:
             raise ValueError('Time step dt is too small')
+        if dt*1e6 % 1 != 0:
```

Comment:
What do you want to check here?

Reply:
As I say in the comment of the PR, we've seen that Parcels unexpectedly fails (outputfiles miss some timeslices) when `dt` has a precision finer than 1e-6. 

So `dt=3.001` works, but `dt=3.0000001` doesn't. 

I want to catch these latter cases

Reply:
Do you have a proposal for a cleaner code to do the same?

Reply:
ah ok sorry. Now is clear. Perhaps just add parenthisis: `(dt * 1e6) % 1`